src/test/java/org/springframework/shell2/DefaultParameterResolverTest.java

```diff
@@ -45,6 +49,58 @@ public void testParses() throws Exception {
 
 	}
 
+	@Test
+	public void testParameterSpecifiedTwiceViaDifferentAliases() throws Exception {
+		Method method = findMethod(Remote.class, "zap", boolean.class, String.class, String.class, String.class);
+
+		thrown.expect(IllegalArgumentException.class);
+		thrown.expectMessage("Named parameter has been specified multiple times via '--bar, --baz'");
+
+		resolver.resolve(
+				makeMethodParameter(method, 0),
+				asList("--force --name --foo y --bar x --baz z".split(" "))
+		);
+	}
+
+	@Test
+	public void testParameterSpecifiedTwiceViaSameKey() throws Exception {
+		Method method = findMethod(Remote.class, "zap", boolean.class, String.class, String.class, String.class);
+
+		thrown.expect(IllegalArgumentException.class);
+		thrown.expectMessage("Parameter for '--baz' has already been specified");
+
+		resolver.resolve(
+				makeMethodParameter(method, 0),
+				asList("--force --name --foo y --baz x --baz z".split(" "))
+		);
+	}
+
+	@Test
+	public void testUnknownParameter() throws Exception {
+		Method method = findMethod(Remote.class, "zap", boolean.class, String.class, String.class, String.class);
+
+		thrown.expect(IllegalArgumentException.class);
+		thrown.expectMessage("Could not lookup parameter for '--unknown' in " + method);
+
+		resolver.resolve(
+				makeMethodParameter(method, 0),
+				asList("--unknown --foo bar".split(" "))
+		);
+	}
+
+	@Test
+	public void testTooMuchInput() throws Exception {
+		Method method = findMethod(Remote.class, "zap", boolean.class, String.class, String.class, String.class);

```

Comment:
Unix commands e.g. tend to do as much as they can.

```shell
➜  ~  ls . whatever
ls: whatever: No such file or directory
.:
Applications [...]
```
`ls whatever` fails but does not prevent `ls .` from being executed.
I'm OK with keeping the current behavior, I was just wondering if it was the most natural one.